recipes-devtools/go/go/0008-fix-filter-compile-flags.patch

```diff
@@ -0,0 +1,34 @@
+From 47c4684c079addf8fc9083e4239e044c612ce3b9 Mon Sep 17 00:00:00 2001
+From: Yevgen Abramov <Yevgen_Abramov@epam.com>
+Date: Mon, 3 Oct 2022 19:50:42 +0300
+Subject: [PATCH] fix filter compile flags
+
+Signed-off-by: Yevgen Abramov <Yevgen_Abramov@epam.com>
+---
+ src/cmd/go/internal/work/exec.go | 7 +++++--
+ 1 file changed, 5 insertions(+), 2 deletions(-)
+
+diff --git a/src/cmd/go/internal/work/exec.go b/src/cmd/go/internal/work/exec.go
+index 63ace6de76..7fe323a7f5 100644
+--- a/src/cmd/go/internal/work/exec.go
++++ b/src/cmd/go/internal/work/exec.go
+@@ -2709,11 +2709,14 @@ func envList(key, def string) []string {
+ var filterFlags = os.Getenv("CGO_PEDANTIC") == ""
+ 
+ func filterCompilerFlags(flags []string) []string {
+-	var newflags []string
++	if len(flags) == 0 {
```

Comment:
Can be combined with the following condition:
```go
if !filterFlags || len(flags) == 0 {
```

Reply:
fixed